storage/tianmu/core/item_tianmu_field.h

```diff
@@ -133,10 +134,12 @@ class Item_tianmudecimal : public Item_decimal {
   Item_tianmudecimal(DataType t);
 
   void Set(int64_t val);
+  String *val_str(String *) override;
 
  private:
   int scale;
   int64_t scaleCoef;
+  std::mutex mtx;
```

Comment:
does the mutex need to be initialized and destroyed?

Reply:
std::scoped_lock manages the life cycle of std::mutex


```

When a scoped_lock object is created, it attempts to take ownership of the mutexes it is given. When control leaves the scope in which the scoped_lock object was created, the scoped_lock is destructed and the mutexes are released. If several mutexes are given, deadlock avoidance algorithm is used as if by [std::lock](https://en.cppreference.com/w/cpp/thread/lock).

```


https://en.cppreference.com/w/cpp/thread/scoped_lock